### PR TITLE
Update Version and Add Latest Field

### PR DESCRIPTION
**Pull request by petedejoy**
To address https://github.com/astronomer/issues/issues/1021

This PR changes the `version` field to match the schema @ashb, @paolaperaza and I agreed upon earlier this week. It also adds a `latest` field so we can dynamically populate the website's download pages with the latest build on a specific version. Each version we update should always have one build where `"latest": true`.

I'm not sure if this breaks anything, but let me know your thoughts @ashb @kaxil!

### CODE REVIEW

**Comment by petedejoy** [comment]
I'd also be open to changing the nested `version` field to `build` so we can differentiate between which version of airflow this distro represents from the actual astro build of that version.

**Comment by ashb** [comment]
Changing the nested field needs a change to the version-check plugin so that isn't as easy.

File: src/astronomer-cea [hunk 1]
```diff
@@ -1,23 +1,26 @@
 {
-  "version": "1.0",
+  "version": "1.10.7",
```
Comment (ashb) [inline src/astronomer-cea]: This is the version of the file, so this shouldn't be changed.

**Comment by petedejoy** [comment]
Hmm ok. Here's what I need for the website to be updated with the right info in a way that will be approachable for users:
 - A field that gives me the Airflow version (ie. 1.10.7)
 - A field that tells me which build number the object maps to (ie. astro.6
 - A field that tells me which one is the latest build number _for that given Airfow version_ so we can automatically generate the appropriate commands and image tags for new users without a ton of manual maintenance.

@ashb - what do you think the best/most plausible way to get this stuff into the JSON is?

**Comment by ashb** [comment]
> - A field that gives me the Airflow version (ie. 1.10.7)
You want to know the latest OSS airflow version? 
> - A field that tells me which build number the object maps to (ie. astro.6)

Do you care about the build number, or do you just want the version to pass in to the `pip install` command?

> - A field that tells me which one is the latest build number for that given Airfow version so we can automatically generate the appropriate commands and image tags for new users without a ton of manual maintenance.

Is this one per "Airflow release series"?

**Comment by petedejoy** [comment]
>Do you care about the build number, or do you just want the version to pass in to the pip install command?

I'd like to list all builds as part of the version information on the page, so it'd be good to have.

For installation, what happens if you don't pass a build number to pip or the docker image? I'd prefer to always give the latest stuff.

> Is this one per "Airflow release series"?

Yes, we have separate pages built for each version of Airflow we cut. Ie. 1.10.7 will have its own download page with the available builds, release notes, etc.

**Comment by ashb** [comment]
> > Do you care about the build number, or do you just want the version to pass in to the pip install command?
> 
> I'd like to list all builds as part of the version information on the page, so it'd be good to have.
> 
> For installation, what happens if you don't pass a build number to pip or the docker image? I'd prefer to always give the latest stuff.

For docker you have to pass "a version" (tag, the bit after `:`) - but we can overwrite tags, so we could have `ap-airflow:1.10.7+astro.7-buster-onbuild` and `ap-airflow:1.10.7-buster-onbuild` (not sure if `+` is allowed in docker tags).

If you did `pip install --upgrade astronomer-certified` without any version specifier it would get the absolute latest. I think if we do `pip install --upgrade astronomer-certified==1.10.7` it would get the latest in the 1.10.7 "series". We'd need to test this pip behaviour though, as I'm not 100% certain of it.

> 
> > Is this one per "Airflow release series"?
> 
> Yes, we have separate pages built for each version of Airflow we cut. Ie. 1.10.7 will have its own download page with the available builds, release notes, etc.

**Comment by petedejoy** [comment]
Ok, so for now while I get this downloads page fetching data and live, it'd be great to have all of the potential info we'll need to build the pages off of this JSON. Do you think a nested `websiteData` object on each release would be overkill?

**Comment by petedejoy** [comment]
Here are some screenshots to add context.

For this page, I need a top-level `Version` (preferably above `available_releases`) that tells me the Airflow/AC version:
image

For this page, I need a list of builds:
image

**Comment by ashb** [comment]
Hmm, that list of "available builds" could potentially get quite long.

How about this as a structure:


```json
{
  "version": "1.0",
  "latest_release_series": "1.10.7",
  "available_releases": [
    {
      "version": "1.10.7-5",
      "level": "bug_fix",
      "url": "https://github.com/astronomer/airflow/releases/tag/1.10.7-5",
      "release_date": "2020-01-12T00:00:00+00:00",
      "release_series": "1.10.7"
    },
    {
      "version": "v1.10.7+astro.6",
      "level": "bug_fix",
      "url": "https://github.com/astronomer/airflow/releases/tag/v1.10.7%2Bastro.6",
      "release_date": "2020-02-21T10:26:00+00:00",
      "release_series": "1.10.7"
    },
    {
      "version": "v1.10.7+astro.7",
      "level": "bug_fix",
      "url": "https://github.com/astronomer/airflow/releases/tag/v1.10.7%2Bastro.7",
      "release_date": "2020-03-12T20:03:00+00:00",
      "release_series": "1.10.7"
    }
  ]
}
```

It requires a small bit of processing on what ever is building that page (to find the latest-by-date in a release_series) but also means we don't have to update old records when we release a new one.

**Comment by petedejoy** [comment]
So I actually don't need the latest release series at the moment- just need latest build (which I guess we could just get by checking the release dates on the client). Could we just have a top -level `available_release_series` array?

```{
  "version": "1.0",
  "available_release_series": ["1.10.7"],  // Can add more here as we build them for the website's central downloads list
  "available_releases": [
    {
      "version": "1.10.7-5",
      "level": "bug_fix",
      "url": "https://github.com/astronomer/airflow/releases/tag/1.10.7-5",
      "release_date": "2020-01-12T00:00:00+00:00",
      "release_series": "1.10.7"
    },
    {
      "version": "v1.10.7+astro.6",
      "level": "bug_fix",
      "url": "https://github.com/astronomer/airflow/releases/tag/v1.10.7%2Bastro.6",
      "release_date": "2020-02-21T10:26:00+00:00",
      "release_series": "1.10.7"
    },
    {
      "version": "v1.10.7+astro.7",
      "level": "bug_fix",
      "url": "https://github.com/astronomer/airflow/releases/tag/v1.10.7%2Bastro.7",
      "release_date": "2020-03-12T20:03:00+00:00",
      "release_series": "1.10.7"
    }
  ]
}```

**Comment by ashb** [comment]
We don't need this for now, so okay to close this?

**Comment by petedejoy** [comment]
Yeah, we can close.

**Pull request closed without merging**